docs/templates/tutorials/subclass-tuner.md

```diff
@@ -0,0 +1,132 @@
+# Subclassing Tuner for Custom Training Loops
+
+The `Tuner` class at `kerastuner.engine.tuner.Tuner` can be subclassed to support advanced uses such as:
+
+- Custom training loops (GANs, reinforement learning, etc.)
+- Adding hyperparameters outside of the model builing function (preprocessing, data augmentation, test time augmentation, etc.)
+
+This tutorial will not cover subclassing to support non-Keras models. To accomplish this, you can subclass the `kerastuner.engine.base_tuner.BaseTuner` class (See `kerastuner.tuners.sklearn.Sklearn` for an example).
+
+### Understanding the search process.
+
+`Tuner.search` can be passed any arguments. These arguments will be passed directly to `Tuner.run_trial`, along with a `Trial` object that contains information about the current trial, including hyperparameters and the status of the trial. Typically, `Tuner.run_trial` is the only method that users need to override when subclassing `Tuner`.
```

Comment:
Besides the detailed example below, I would recommend having a sketch of the code here, e.g.

```python

class MyTuner(kt.Tuner):

    def run_trial(self, trial, ...):
        hp = trial.hyperparameters
        model = self.hypermodel.build(trial.hyperparameters)
        for epoch in range(10):
              epoch_loss = ...
              self.on_epoch_end(trial, model, epoch, logs={'loss': epoch_loss})
```

Otherwise it's hard to parse what's essential and what's not.

Reply:
Done